[0liverkong] iP

0liverkong opened the pull request:
### Dook 🐈

>  Meow :3

You should use Dook if you

1. Love cats
2. I can't think of anything else, sorry

It can

- **Dance**
- _Meow_
- _**Be cute**_
- and do many way less important things (like manage your tasks)

Checklists

- [ ] I love checklists!

[The source for the emoticons!](https://2ch-aa.blogspot.com/2018/06/625.html)

To run, write `java -jar Dook.jar` in your command line :3

some java code
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Completed Week 2 Level 0

Commit:
Completed Week 2 Level 1

Commit:
Completed Week 2 Level 2

Commit:
Completed Week 2 Level 3

0liverkong committed:
Completed Week 2 Level 4

0liverkong committed:
Completed Week 2 Level 5

0liverkong committed:
completed Week 2 Level 6

0liverkong committed:
Small cleanup

0liverkong committed:
Fix typo

0liverkong committed:
Completed Week 2 TextUITesting

0liverkong committed:
Change indentation to match style convention

0liverkong committed:
Separate Dook and Ui classes

0liverkong committed:
Create Command and Parser classes

0liverkong committed:
Add OOP functionality for Command and Parser classes and subclasses

0liverkong committed:
Add functionality to save on bye command

0liverkong committed:
Add file reading and writing functionality and more OOP abstraction

0liverkong committed:
Add auto-creation of /data directory if not present

0liverkong committed:
Add auto-creation of /data directory if not present

0liverkong committed:
Add datetime functionality

0liverkong committed:
Add package for tasks

0liverkong committed:
Add package for dook and tasks

0liverkong committed:
Add package for dook and tasks

0liverkong committed:
Merge branch 'master' into add-gradle-support

0liverkong committed:
Add Gradle support

0liverkong committed:
Merge branch 'add-gradle-support'

0liverkong committed:
Add JUnit tests

0liverkong committed:
Add packaging functionality

0liverkong committed:
Add half of the JavaDoc messages

0liverkong committed:
Merge branch 'branch-A-JavaDoc'

0liverkong committed:
Tweak code to match Java coding standards

0liverkong committed:
Merge tag 'A-CodingStandard'

0liverkong committed:
Add functionality to find tasks based on a keyword

0liverkong committed:
Add meowing

0liverkong committed:
Add dancing

ChuaZiLong commented on src/main/java/Dook/Parser.java:
```diff
@@ -0,0 +1,199 @@
+import java.util.Arrays;
+import java.time.LocalDateTime;
+
+public class Parser {
```
I like that the class names are in noun form.

ChuaZiLong commented on src/main/java/Dook/Parser.java:
```diff
@@ -0,0 +1,199 @@
+     * @throws DookException If an invalid command is detected.
+     */
+    public static Command parse(String input) throws DookException {
+        String[] split = input.split(" ", 2);
```
I like that the variable names use standard words.

ChuaZiLong commented on src/main/java/Dook/Parser.java:
```diff
@@ -0,0 +1,199 @@
+     */
+    public static Command parse(String input) throws DookException {
+        String[] split = input.split(" ", 2);
+        String firstWord = split[0];
```
I like that the variable names explain what it is to the reader.

ChuaZiLong commented on src/main/java/Dook/Parser.java:
```diff
@@ -0,0 +1,199 @@
+     */
+    public static Task parseFileLineToTask(String s) throws DookException {
+        String[] split = s.split(" \\| ", 3);
+        String taskTypeString = split[0];
```
I like that the names are neither too long nor too short.

ChuaZiLong commented on src/main/java/Command/FindCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ArrayList<Task> found = new ArrayList<>();
```
Perhaps it would be good to rename this to make it clear that it is a multi-valued variable.

ReganChoy commented on src/main/java/Command/Command.java:
```diff
@@ -0,0 +1,24 @@
+import Dook.Storage;
+import Dook.DookException;
+
+public abstract class Command {
```
Interesting use of an abstract class

ReganChoy commented on src/main/java/Command/DeleteCommand.java:
```diff
@@ -0,0 +1,46 @@
+import Dook.DookException;
+
+public class DeleteCommand extends Command {
+    private final int positionToDelete;
```
Is this meant to be a constant? Should this be in all caps?

ReganChoy commented on src/main/java/Command/FindCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+public class FindCommand extends Command {
+
+    private final String target;
```
Same thing as before, is this a constant?

ReganChoy commented on src/main/java/Dook/Parser.java:
```diff
@@ -0,0 +1,199 @@
+import Task.TaskType;
+import Task.ToDo;
+import Task.Task;
+
```
Very good use of abstraction and OOP!!

ReganChoy commented on src/main/java/Dook/Storage.java:
```diff
@@ -0,0 +1,88 @@
+
+public class Storage {
+
+    private final ArrayList<String> path;
```
Since ArrayList is under Collections, maybe path should be renamed into paths?

ReganChoy commented on src/main/java/Dook/Ui.java:
```diff
@@ -0,0 +1,69 @@
+            "　　　 ∩・ω・｀）\n" +
+            "　　　 |　　 ⊂ﾉ\n" +
+            "　　 　｜　　 _⊃　　♪\n" +
+            "　　　 し ⌒,"));
```
That is a very nice cat. Should the + operators be in front of every line?

0liverkong committed:
Add skeleton for JavaFX

0liverkong committed:
Make master branch working CLI

0liverkong committed:
Add echo functionality for GUI skeleton

0liverkong committed:
Add partially working GUI

0liverkong committed:
Add partially working GUI

0liverkong committed:
Add animations, and app closing

0liverkong committed:
Deconflict commits

0liverkong committed:
Deconflict more files

0liverkong committed:
Fix some checkstyles

0liverkong committed:
Fix some coding style

0liverkong committed:
Merge branch 'branch-A-Checkstyle'

0liverkong committed:
Add more checkstyle

0liverkong committed:
Merge branch 'branch-A-Checkstyle'

0liverkong committed:
Improve code quality

Code quality is important as it allows for error-resistant coding,
and a standardised style that is shared throughout all the
contributors to a project.

0liverkong committed:
Add new task type DoAfter

This task type allows the user to schedule tasks that have to be
done after a certain date - the opposite of a deadline.

0liverkong committed:
Improve code quality in MainWindow.java

0liverkong committed:
Merge branch 'branch-BCD-Extension'

0liverkong committed:
Fix typos

0liverkong committed:
Add final changes to code, add more Javadocs

0liverkong committed:
Add more test cases for JUnit

0liverkong committed:
Add more test cases for JUnit for DeadlineTest

0liverkong committed:
Update README

0liverkong committed:
Add product picture

0liverkong committed:
Move readme into docs

0liverkong committed:
Adjust response timing
